pkg/controller/operators/olm/operator.go

```diff
@@ -456,48 +492,61 @@ func (a *Operator) syncClusterServiceVersion(obj interface{}) (syncError error)
+	if !(outCSV.Status.LastUpdateTime == clusterServiceVersion.Status.LastUpdateTime &&
 		outCSV.Status.Phase == clusterServiceVersion.Status.Phase &&
 		outCSV.Status.Reason == clusterServiceVersion.Status.Reason &&
-		outCSV.Status.Message == clusterServiceVersion.Status.Message {
-		return
-	}
+		outCSV.Status.Message == clusterServiceVersion.Status.Message) {
 
-	// Update CSV with status of transition. Log errors if we can't write them to the status.
-	updatedCSV, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(clusterServiceVersion.GetNamespace()).UpdateStatus(outCSV)
-	if err != nil {
-		updateErr := errors.New("error updating ClusterServiceVersion status: " + err.Error())
-		if syncError == nil {
-			logger.Info(updateErr)
-			return updateErr
+		// Update CSV with status of transition. Log errors if we can't write them to the status.
+		_, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(outCSV.GetNamespace()).UpdateStatus(outCSV)
+		if err != nil {
+			updateErr := errors.New("error updating ClusterServiceVersion status: " + err.Error())
+			if syncError == nil {
+				logger.Info(updateErr)
+				return updateErr
+			}
+			syncError = fmt.Errorf("error transitioning ClusterServiceVersion: %s and error updating CSV status: %s", syncError, updateErr)
 		}
-		syncError = fmt.Errorf("error transitioning ClusterServiceVersion: %s and error updating CSV status: %s", syncError, updateErr)
 	}
 
-	operatorGroup := a.operatorGroupForActiveCSV(logger, updatedCSV)
+	operatorGroup := a.operatorGroupForActiveCSV(logger, outCSV)
 	if operatorGroup == nil {
 		logger.WithField("reason", "no operatorgroup found for active CSV").Info("skipping CSV resource copy to target namespaces")
 		return
 	}
 
-	if updatedCSV.Status.Phase == v1alpha1.CSVPhaseFailed && updatedCSV.Status.Reason == v1alpha1.CSVReasonInterOperatorGroupOwnerConflict {
-		logger.WithField("reason", updatedCSV.Status.Message).Info("skipping CSV resource copy to target namespaces")
-		return
+	allNamespaces := make([]string, 0)
+	pruneNamespaces := make([]string, 0)
+	targetNamespaces := make([]string, 0)
+	namespaceObjs, err := a.lister.CoreV1().NamespaceLister().List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range namespaceObjs {
+		allNamespaces = append(allNamespaces, ns.GetName())
+	}
+	if len(operatorGroup.Status.Namespaces) == 1 && operatorGroup.Status.Namespaces[0] == corev1.NamespaceAll {
+		targetNamespaces = allNamespaces
+	} else {
+		targetNamespaces = operatorGroup.Status.Namespaces
+		pruneNamespaces = sliceCompare(allNamespaces, targetNamespaces)
+		logger.Debugf("Found namespaces to clean %v", pruneNamespaces)
 	}
 
 	// Check if we need to do any copying / annotation for the operatorgroup
-	if err := a.copyCsvToTargetNamespace(updatedCSV, operatorGroup); err != nil {
+	if err := a.copyCsvToTargetNamespace(clusterServiceVersion, operatorGroup, targetNamespaces, pruneNamespaces); err != nil {
```

Comment:
We want to make sure the CSV isn't failed due to an intersecting `OperatorGroup` issue before copying to namespaces with intersecting provided APIs. As it stands, I think this breaks intersection logic because copied CSVs can stomp on good ones.